src/views/Settings.vue

```diff
@@ -345,18 +395,60 @@ export default {
   methods: {
     ...mapActions(['notify', 'send', 'getSpaces']),
     async handleSubmit() {
-      this.loading = true;
-      try {
-        await this.send({
-          space: this.key,
-          type: 'settings',
-          payload: this.form
-        });
-      } catch (e) {
-        console.log(e);
+      if (this.isValid) {
+        this.loading = true;
+        try {
+          await this.send({
+            space: this.key,
+            type: 'settings',
+            payload: this.form
+          });
+        } catch (e) {
+          console.log(e);
+        }
+        await this.getSpaces();
+        this.loading = false;
+      } else {
+        this.showErrors = true;
+      }
+    },
+    inputError(field) {
+      if (!this.isValid && !this.loading && this.showErrors) {
+        const required = this.validate.find(
+          error =>
+            error.keyword === 'required' &&
+            error.params.missingProperty === field
+        );
+        if (required) return this.$t('errors.fieldRequired');
+
+        const minLength = this.validate.find(
```

Comment:
There is a lot of code to map each of the errors, can't we just check if default.json>errors[field] exist to show error or not. ?

Reply:
Yeah, I'll try to come up with something more compact tomorrow.